crates/iceberg/src/scan.rs

```diff
@@ -503,17 +544,36 @@ pub struct FileScanTask {
     start: u64,
     #[allow(dead_code)]
     length: u64,
+    project_field_id: Vec<usize>,
+    predicate: Option<BoundPredicate>,
```

Comment:
We might want to consider annotating this line with `#[serde(skip_serializing_if = "Option::is_none")]` perhaps to save space in the serialized tasks when performing a full scan